### PR TITLE
CORE-7511 Add more user info to PermID Request details.

### PR DESCRIPTION
**Pull request by psarando**
Updated the responses of Permanent ID Request endpoints that return a request's details to include more user info in the 'requested_by' field, such as email and first/last name.
Since the requesting user's email is now available when notifications are sent to that user, notification emails can now be enabled.
Added the new ID to the "Completion" notification update in the "comments" field, which is now included as the body of the "Completion" notification email.

### CODE REVIEW

File: services/terrain/src/terrain/services/permanent_id_requests.clj [hunk 1]
```diff
@@ -263,6 +269,19 @@
       (dissoc :target_id :target_type :original_path)
       (assoc :folder (data-info/stat-by-uuid user (uuidify target_id)))))
 
+(defn- format-requested-by
+  [user {:keys [requested_by target_id] :as permanent-id-request}]
+  (let [user-info (groups/lookup-subject user requested_by)]
+    (if user-info
```
Comment (ianmcorvidae) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: This is, of course, minor, but may as well use `if-let` here.

File: services/terrain/src/terrain/services/permanent_id_requests.clj [hunk 2]
```diff
@@ -302,25 +326,25 @@
...
+       (format-permanent-id-request-details (:shortUsername current-user))))
 
 (defn update-permanent-id-request
   [request-id params body]
   (let [response (->> (metadata/update-permanent-id-request request-id body)
                       parse-service-json
-                      (format-perm-id-req-response (:shortUsername current-user)))]
+                      (format-permanent-id-request-details (:shortUsername current-user)))]
     (send-update-notification response)
     response))
```
Comment (ianmcorvidae) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: This isn't something you're changing here, but am I missing something or is `params` not used in this function? If so, why is it included?

File: services/terrain/src/terrain/services/permanent_id_requests.clj [hunk 3]
```diff
@@ -355,4 +379,5 @@
   (let [identifier (complete-permanent-id-request (:shortUsername current-user)
                                                   (admin-get-permanent-id-request request-id nil))]
     (update-permanent-id-request request-id nil (json/encode {:status       status-code-completion
+                                                              :comments     identifier
```
Comment (ianmcorvidae) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: This seems unrelated (though presumably fine) -- what's going on here adding the extra key with the same info?

Comment (ianmcorvidae) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: Oh, I see, this is what's going into the body of the email. Why not just use the pre-existing `:permanent_id` key, then?

File: services/terrain/src/terrain/services/permanent_id_requests.clj (hunk 1 above)
Comment (psarando) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: Thanks, good catch.

File: services/terrain/src/terrain/services/permanent_id_requests.clj (hunk 2 above)
Comment (psarando) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: No, `params` is not used. It was added at the beginning to all these functions when they were stubbed out in case they were needed, but the params never got used.

File: services/terrain/src/terrain/services/permanent_id_requests.clj (hunk 3 above)
Comment (psarando) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: This status update is an automated `Completion` status update.

The `comments` field can be any string and is the comments of the status update, which is displayed in the request's status history and also used for the body of the status update's notification. So the comments also become the body of the notification's email.
Other status updates besides the `Completion` status can set any other arbitrary string for the comments, and those mainly come from the UI when an admin is updating the status of a request.

The `permanent_id` is the newly created ID. This field is specific to the `POST /admin/permanent-id-requests/{request-id}/status` metadata endpoint and is only sent once when the request is completed, then the metadata service saves the ID with the Permanent ID Request records. This endpoint will return an error if this field is populated when an ID is already saved for the Permanent ID Request.

Comment (ianmcorvidae) [inline services/terrain/src/terrain/services/permanent_id_requests.clj]: Ah, alright. Makes sense; I'd somehow missed other places setting this value.

**Comment by ianmcorvidae** [comment]
Changes look good, I'd say. It wouldn't hurt to refactor to `if-let` or remove that unused function argument but it's also fine without that. :shipit:
